allennlp/common/file_utils.py

```diff
@@ -92,6 +95,73 @@ def cached_path(url_or_filename: Union[str, Path], cache_dir: str = None) -> str
         raise ValueError("unable to parse {} as a URL or as a local path".format(url_or_filename))
 
 
+def split_s3_path(url: str) -> Tuple[str, str]:
+    """Split a full s3 path into the bucket name and path."""
```

Comment:
would it be better to use `urlparse` like here: https://stackoverflow.com/a/42641363/1076346 ?

(I don't know, it's just that logic like "split on slashes and take the third and fourth parts"  makes me nervous)

Reply:
Got it, just changed that